server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -392,24 +401,31 @@ protected void doRun() {
                 assert initializingSnapshots.contains(snapshot.snapshot());
                 Repository repository = repositoriesService.repository(snapshot.snapshot().getRepository());
 
-                MetaData metaData = clusterState.metaData();
-                if (!snapshot.includeGlobalState()) {
-                    // Remove global state from the cluster state
-                    MetaData.Builder builder = MetaData.builder();
-                    for (IndexId index : snapshot.indices()) {
-                        builder.put(metaData.index(index.getName()), false);
+                if (clusterState.nodes().getMinNodeVersion().onOrAfter(NO_REPO_INITIALIZE_VERSION)) {
+                    final String snapshotName = snapshot.snapshot().getSnapshotId().getName();
```

Comment:
Copy-paste of part of `initializeSnapshot` method. Either remove this code from `initializeSnapshot` and call it here for both cases or create `checkPreconditions` method.

Reply:
Yea makes sense on second though -> removed the checks from `BlobStoreRepository`